qcodes_contrib_drivers/drivers/SingleQuantum/SingleQuantum.py

```diff
@@ -0,0 +1,498 @@
+                value=t_in_ms))
+        self.talk.send(msg)
+
+    def get_number_of_detectors(self):
+        return self.talk.get_label("NumberOfDetectors")["value"]
+
+    def get_measurement_periode(self):
+        """Get measurment periode in ms.
+        Return (float): time
+        """
+        return self.talk.get_label("InptMeasurementPeriod")["value"]
+
+    def get_bias_current(self):
+        return self.talk.get_label("BiasCurrent")["value"]
+
+    def get_trigger_level(self):
+        return self.talk.get_label("TriggerLevel")["value"]
+
+    def get_bias_voltage(self):
+        msg = json.dumps(dict(request="BiasVoltage"))
+        self.talk.send(msg)
+        return self.talk.get_label("BiasVoltage")["value"]
+
+    def set_bias_current(self, current_in_uA):
+        array = current_in_uA
+        msg = json.dumps(dict(command="SetAllBiasCurrents",
+                              label="BiasCurrent", value=array))
+        self.talk.send(msg)
+
+    def set_trigger_level(self, trigger_level_mV):
+        array = trigger_level_mV
+        msg = json.dumps(dict(command="SetAllTriggerLevels",
+                              label="TriggerLevel", value=array))
+        self.talk.send(msg)
+
+    def enable_detectors(self, state=True):
+        msg = json.dumps(dict(command="DetectorEnable",
+                              label="DetectorEnable", value=state))
+        self.talk.send(msg)
+
+    def set_dark_counts_auto_iv(self, dark_counts):
+        """Set the dark counts for the automatic detector calibration.
+        After this command execute: self.auto_cali_bias_currents()
+        """
+        if self.NUMBER_OF_DETECTORS != len(dark_counts):
+            raise ValueError(
+                'Dark counts not the same lenght as number of detectors')
+        else:
+            msg = json.dumps(
+                dict(
+                    command="DarkCountsAutoIV",
+                    label="DarkCountsAutoIV",
+                    value=dark_counts))
+            self.talk.send(msg)
+
+    def auto_cali_bias_currents(self):
+        """Performs the automatic bias calibration. Make sure that no light reaches the detectors during this procedure.
+        To check if this function has finished use self.auto_cali_finished() which return true or false.
+        """
+        msg = json.dumps(
+            dict(
+                command="AutoCaliBiasCurrents",
+                value=True))
+        self.talk.send(msg)
+
+    def auto_cali_finished(self):
+        """Check if auto calibration of the bias currents to find a given dark count value has finished.
+        Returns: True if finished, False otherwise
+        """
+        msg = json.dumps(dict(request="StartAutoIV"))
+        self.talk.send(msg)
+        return not(self.talk.get_label("StartAutoIV")["value"])
+
+    def error(self, error_msg):
+        """Called in case of an error"""
+        print("ERROR DETECTED")
+        print(error_msg)
```

Comment:
I suggest using logging as well or instead of print for handling error messages. All qcodes drivers have a `self.log` standard python logging module attached

Reply:
Implemented suggestion.